xmppserver/src/main/java/org/jivesoftware/openfire/muc/MUCRoom.java

```diff
@@ -1308,6 +1308,48 @@ public void removeOccupant(@Nonnull final MUCOccupant occupant) {
         MUCEventDispatcher.occupantLeft(occupant.getOccupantJID(), occupant.getUserAddress(), occupant.getNickname());
     }
 
+    /**
+     * Clears the chat history of a room. Sends out message retraction stanzas to all occupants, for all the message in the chat history.
+     */
+    public void clearChatHistory() {
+        Iterator<Message> history = roomHistory.getMessageHistory();
```

Comment:
This will work in principle, but I think it should be improved in a number of ways:
1. As the generation and distribution of revocation messages is potentially a (very) long running process, it should be async.
2. Lets iterate over the history backwards, sending a revocation for the most recently received message first. That message is a lot more likely to be currently displayed in a connected client, than the oldest message in a chatroom. By working backwards, we start with the most visible messages. I believe that this will lead to a better user-experience.
3. Please add a configuration option (in the form of a SystemProperty) that allows an administrator to skip sending the revocation messages.

Reply:
Thank you Guus. I've updated the code as suggested.